### ISSUE TITLE
constants parsed as type_mark

### ISSUE THREAD

**Issue by Giftkekschen** [jeremiah-c-leary/vhdl-style-guide#1545]
**Environment**
VSG version: 3.35.0
OS: Ubuntu 22.04.5 LTS

**Describe the bug**
When using a constant of an array type in a range definition, the identifier of the constant is parsed as type_mark.

**To Reproduce**

1. Minimal vhdl example:
```vhdl
library ieee;
  use ieee.std_logic_1164.all;
  use ieee.numeric_std.all;

entity test is
end entity test;

architecture rtl of test is

  type integer_array_type is array (natural range <>) of integer;

  constant CONST : integer_array_type := (3, 2, 1);
  signal   test  : std_logic_vector(CONST(0) - 1 downto 0);

begin

end architecture rtl;

```
2. Minimal config:
```yaml
rule:
  global:
    disable: true
  type_mark_500:
    disable: false
  constant_004:
    disable: false
    case: upper

``` 
3. run `vsg -c minimal.yml -f minimal.vhd`
4. vsg reports:
```
----------------+------------+------------+--------------------------------------
  Rule          |  severity  |  line(s)   | Solution
----------------+------------+------------+--------------------------------------
  type_mark_500 | Error      |         13 | Change "CONST" to "const"
----------------+------------+------------+--------------------------------------
```

**Expected behavior**
The indexed constant is not a type_mark, therefore, no errors should be produced. 

**Screenshots**
NA

**Additional context**
NA
